agent/common.py

```diff
@@ -0,0 +1,276 @@
+    Returns:
+        The vulnerability location object.
+    """
+
+    asset: (
+        ipv4_asset.IPv4
+        | ipv6_asset.IPv6
+        | domain_asset.DomainName
+        | file_asset.File
+        | android_aab_asset.AndroidAab
+        | android_apk_asset.AndroidApk
+        | ios_ipa_asset.IOSIpa
+    )
+
+    if selector == "v3.asset.file.android.apk":
+        asset = android_apk_asset.AndroidApk(
+            path=file_path, content=file_content, content_url=content_url
+        )
+        return agent_report_vulnerability_mixin.VulnerabilityLocation(
+            asset=asset,
+            metadata=[
+                agent_report_vulnerability_mixin.VulnerabilityLocationMetadata(
+                    metadata_type=agent_report_vulnerability_mixin.MetadataType.FILE_PATH,
+                    value=file_path or "",
+                )
+            ],
+        )
+    elif selector == "v3.asset.file.android.aab":
+        asset = android_aab_asset.AndroidAab(
+            path=file_path, content=file_content, content_url=content_url
+        )
+        return agent_report_vulnerability_mixin.VulnerabilityLocation(
+            asset=asset,
+            metadata=[
+                agent_report_vulnerability_mixin.VulnerabilityLocationMetadata(
+                    metadata_type=agent_report_vulnerability_mixin.MetadataType.FILE_PATH,
+                    value=file_path or "",
+                )
+            ],
+        )
+    elif selector == "v3.asset.file.ios.ipa":
+        asset = ios_ipa_asset.IOSIpa(
+            path=file_path, content=file_content, content_url=content_url
+        )
+        return agent_report_vulnerability_mixin.VulnerabilityLocation(
+            asset=asset,
+            metadata=[
+                agent_report_vulnerability_mixin.VulnerabilityLocationMetadata(
+                    metadata_type=agent_report_vulnerability_mixin.MetadataType.FILE_PATH,
+                    value=file_path or "",
+                )
+            ],
+        )
+    elif selector == "v3.asset.file":
+        asset = file_asset.File(path=file_path, content=file_content)
+        return agent_report_vulnerability_mixin.VulnerabilityLocation(
+            asset=asset,
+            metadata=[
+                agent_report_vulnerability_mixin.VulnerabilityLocationMetadata(
+                    metadata_type=agent_report_vulnerability_mixin.MetadataType.FILE_PATH,
+                    value=file_path or "",
+                )
+            ],
+        )
+    else:
+        if target_url is None:
+            return None
+        metadata = []
+        target = parse.urlparse(target_url)
+        ip = None
+        port = None
+        potential_ip = target_url
+        if target.scheme != "":
+            potential_ip = potential_ip.replace(f"{target.scheme}://", "")
+        if _is_ipv4(potential_ip) is True:
+            ip, port = _split_ipv4(potential_ip)
+            asset = ipv4_asset.IPv4(host=str(ip), version=4, mask="32")
```

Comment:
isn't it already a string?
```suggestion
            asset = ipv4_asset.IPv4(host=ip, version=4, mask="32")
```